File: src/main/java/org/smudge/petstore/PetController.java
package org.smudge.petstore;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

@RestController()
@RequestMapping("pet")
public class PetController {

    @Autowired
    private PetService petService;

    @GetMapping()
    public Pet getPet(@RequestParam int id) {
        return petService.getPet(id);
    }

    @PostMapping
    public Pet createPet(@RequestBody Pet newPet) {
        return petService.createPet(newPet.getOwnerId(), newPet.getName(), newPet.getType());
    }

    @PutMapping
    public Pet changeName(@RequestParam int id, @RequestParam String newName) {
        return petService.changeName(id, newName);
    }
}

File: src/main/java/org/smudge/petstore/PetController.kt
package org.smudge.petstore

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("pet")
class PetController(
        @Autowired
        private val petService: PetService
) {

    @GetMapping
    fun getPet(@RequestParam id: Int): Pet {
        return petService.getPet(id)
    }

    @PostMapping
    fun createPet(@RequestBody newPet: Pet): Pet {
        return petService.createPet(newPet.id, newPet.ownerId, newPet.name, newPet.type)
    }

    @PutMapping
    fun changeName(@RequestParam id: Int, @RequestParam ownerId:Int, @RequestParam newName: String = "Tony", @RequestParam type:String = "Cat" ): Pet {
        return petService.changeName(id, ownerId, newName, type)
    }
}